src/prop/prop_proof_manager.cpp

```diff
@@ -114,19 +121,192 @@ std::vector<Node> PropPfManager::getUnsatCoreLemmas()
 {
   std::vector<Node> usedLemmas;
   std::vector<Node> allLemmas = getLemmaClauses();
-  std::shared_ptr<ProofNode> satPf = getProof(false);
-  std::vector<Node> satLeaves;
-  expr::getFreeAssumptions(satPf.get(), satLeaves);
+  // compute the unsat core clauses, as below
+  std::vector<Node> ucc = getUnsatCoreClauses();
+  Trace("prop-pf") << "Compute unsat core lemmas from " << ucc.size()
+                   << " clauses (of " << allLemmas.size() << " lemmas)"
+                   << std::endl;
+  Trace("prop-pf") << "lemmas: " << allLemmas << std::endl;
+  Trace("prop-pf") << "uc: " << ucc << std::endl;
+  // filter to only those corresponding to lemmas
   for (const Node& lemma : allLemmas)
   {
-    if (std::find(satLeaves.begin(), satLeaves.end(), lemma) != satLeaves.end())
+    if (std::find(ucc.begin(), ucc.end(), lemma) != ucc.end())
     {
       usedLemmas.push_back(lemma);
     }
   }
   return usedLemmas;
 }
 
+std::vector<Node> PropPfManager::getMinimizedAssumptions()
+{
+  std::vector<Node> minAssumptions;
+  std::vector<SatLiteral> unsatAssumptions;
+  d_satSolver->getUnsatAssumptions(unsatAssumptions);
+  for (const Node& nc : d_assumptions)
+  {
+    if (nc.isConst())
+    {
+      if (nc.getConst<bool>())
+      {
+        // never include true
+        continue;
+      }
+      else
+      {
+        minAssumptions.clear();
+        minAssumptions.push_back(nc);
+        return minAssumptions;
+      }
+    }
+    else if (d_pfCnfStream.hasLiteral(nc))
```

Comment:
what is this case for? The call to getUnsatAssumptions of the SAT solver above will be already doing some filtering, so I'm a bit confused of what is being done here.

Reply:
I believe this is just being safe prior to calling `d_pfCnfStream.getLiteral`, which will segfault if the literal does not exist.

I am not sure if the "else" branch of this should ever fire, I will double check.

Reply:
This indeed should always hold, I will add a debug assertion in the "else" case.